### PR TITLE
feat(uboot): add rock4cplus support

### PR DESCRIPTION
**Pull request by jgero**
## Description of changes

I added the [U-Boot config to build sd images for the Rock 4 C+](https://source.denx.de/u-boot/u-boot/-/blob/HEAD/configs/rock-4c-plus-rk3399_defconfig).

## Things done

I built the SD image I'm currently using on my Rock 4C+ board using this config.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by lopsided98 — CHANGES_REQUESTED** [review]
It looks like the boards are mostly in alphabetical order, so this one should be moved into the proper place as well.

Otherwise looks good.

**Comment by jgero** [comment]
> It looks like the boards are mostly in alphabetical order, so this one should be moved into the proper place as well.
> 
> Otherwise looks good.

Done :)

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -28361,6 +28361,7 @@ with pkgs;
     ubootSopine
     ubootUtilite
     ubootWandboard
+    ubootRock4CPlus
```
Comment (lopsided98) [inline pkgs/top-level/all-packages.nix:28364]: Sorry for continuing to nitpick, but this should be moved alphabetically too.

**Review by lopsided98 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (jgero) [reply pkgs/top-level/all-packages.nix]: My bad, forgot that one :facepalm:

**Review by lopsided98 — APPROVED** [review]
Don't have hardware to test, but looks good.

**Review by 0x4A6F — APPROVED** [review]
Thanks, builds successfully for native `aarch64-linux` and `pkgsCross.aarch64-multiplatform.ubootRock4CPlus`.

**Pull request merged**